Added Jacoco coverage in build

Pietro-MadeTech opened the pull request.

jdwoolcock committed:
Added Jacoco coverage

Coverage report is triggered at build time.

Both xml output and html output are generated for the reports

xml output is used by GH actions

jdwoolcock reviewed (changes requested):
Not quite sure as why but `./gradlew clean build`  would normally run 115 tests  however with this change it is only running 53.

mattonfoot commented:
@jdwoolcock @Pietro-MadeTech reinstated all tests to show actual coverage

jdwoolcock commented on README.md:
```diff
@@ -66,6 +66,11 @@ will also reset the database
+
+## Coverage 
+Coverage report is generated by Jacoco.
+After running the tests the coverage reports are in `build/reports/jacoco`
```
Nice! :sparkles:

Might be good to extend this to include how to show in intelliJ
There maybe a better way, but this worked for me


- Go to **Run** in the main menu.
- Select **Manage Coverage Reports**
- Navigate to and select the JaCoCo execution data file
- Click **OK**. The coverage report will now appear in the **Coverage Tool Window** and the code will be highlighted in the editor.

Pietro-MadeTech committed:
Added instruction to add and open report in intellij/browser

Pietro-MadeTech committed:
Created CI/CD workflow for coverage
The workflow sends a comment with the coverage percentages on the PR

It parses and calculates the coverage by excluding not required files to match the html report

jdwoolcock reviewed (approved):
LGTM 🎉
